### PR TITLE
fix: pprof ParseToPprof

### PR DESCRIPTION
**Pull request by korniltsev**
address https://github.com/grafana/pyroscope/issues/2359

### CODE REVIEW

File: pkg/og/convert/pprof/profile.go [hunk 1]
```diff
@@ -275,3 +129,69 @@ func (p *RawProfile) loadPprofFromForm() error {
...
+	}
+	if strings.Contains(st, "alloc_") || strings.Contains(st, "inuse_") {
+		return "memory"
+	}
+	if strings.Contains(st, "mutex_") {
+		return "mutex"
+	}
+	if strings.Contains(st, "mutex_") {
+		return "mutex"
+	}
```
Comment (kolesnikovae) [inline pkg/og/convert/pprof/profile.go:154]:
```suggestion
	if strings.Contains(st, "mutex_") {
		return "mutex"
	}
```

File: pkg/og/convert/pprof/profile.go [hunk 2]
```diff
@@ -259,10 +109,14 @@ func (p *RawProfile) loadPprofFromForm() error {
...
 	}
-	p.PreviousProfile, err = form.ReadField(f, formFieldPreviousProfile)
+	PreviousProfile, err := form.ReadField(f, formFieldPreviousProfile)
 	if err != nil {
 		return err
 	}
+	if PreviousProfile != nil {
+		return fmt.Errorf("cumulative profile is nolonger supported. " +
+			"Update clients to the latest versions of pyroscope-golang/godeltaprof")
+	}
```
Comment (kolesnikovae) [inline pkg/og/convert/pprof/profile.go:119]: I guess `godeltaprof` is our internal dependency users might not be aware of

```suggestion
	if PreviousProfile != nil {
		return fmt.Errorf("unsupported client version. " +
			"Please update github.com/grafana/pyroscope-go to the latest version")
	}
```

**Review by kolesnikovae — APPROVED** [review]
🎉

**Comment by korniltsev** [comment]
Btw this breaks all the rideshare examples
cc @grafana/pyroscope-team

**Comment by korniltsev** [comment]
@kolesnikovae ptal again. I've adde two fixes since your last review. 
1 - there was a wrong time in pyspy pprofs
2 - I had to rewrite function names for pyspy and rbspy to include function name. There used to be a line number in the past, I decided not to add it for now.

**Comment by korniltsev** [comment]
pyspy issue https://github.com/grafana/pyroscope-rs/pull/134

**Comment by korniltsev** [comment]
o_O I did not expect automerge with no second approve

**Comment by kolesnikovae** [comment]
The new changes look reasonable 👍🏻

**Comment by kolesnikovae** [comment]
> Btw this breaks all the rideshare examples

I was under impression that only Go sends delta profiles (not sure about nodejs), which would indeed stop working. But the example in thte repo (go-push) already uses pyroscope-go with `deltagoprof`, and AFAIK apart from that we don't have anything. Old rideshare examples (demo.pyroscope.io) should not be affected because in pyroscope cloud we use the old parser. Am I missing something?

**Comment by korniltsev** [comment]
the dynamic tags( for example vehicle in rideshare or [this](https://github.com/grafana/pyroscope/blob/main/examples/python/simple/main.py) ) will be broken, not the delta profiles

**Pull request merged**